lib/api/api.js

```diff
@@ -1957,27 +1960,29 @@ api.prototype.isReattachable = function(inputAddresses, callback) {
 /*
  * Wraps {checkConsistency} in a promise so that its value is returned
  */
-api.prototype.isPromotable = function(tail) {
+api.prototype.isPromotable = function(tail, options) {
     var self = this;
 
+    if (!options) options = {}
+
     // Check if is hash
     if (!inputValidator.isHash(tail)) {
         return false;
     }
 
     var command = apiCommands.checkConsistency([tail]);
 
-    var promise = new Promise(function(res, rej) {
-        self.sendCommand(command, function(err, isConsistent) {
+    return new Promise(function(resolve, reject) {
+        self.sendCommand(command, function(err, res) {
             if (err) {
-              rej(err)
+              reject(err)
             }
-            res(isConsistent);
+            if (!res.state && options.rejectWithReason) {
```

Comment:
Would it be better for this API to return both `state` and `info`? I guess it would make sense to have a separate method for `checkConsistency` that returns the original response `{state, info}`.

Reply:
That would be a breaking change, preferably to be avoided, hence I added that option to reject with info. We can continue discussion in #251